tianshou/policy/modelbase/psrl.py

```diff
@@ -0,0 +1,187 @@
+        rew_mean_prior: np.ndarray,
+        rew_std_prior: np.ndarray,
+    ) -> None:
+        self.trans_count = trans_count_prior
+        self.trans_count_prior_sum = np.sum(trans_count_prior, axis=2)
+        self.n_action, self.n_state, _ = trans_count_prior.shape
+        self.rew_mean = rew_mean_prior
+        self.rew_std = rew_std_prior
+        self.rew_count = np.ones_like(rew_mean_prior)
+        self.policy: Optional[np.ndarray] = None
+        self.updated = False
+
+    def observe(
+        self, trans_count: np.ndarray,
+            rew_sum: np.ndarray, rew_count: np.ndarray
+    ) -> None:
+        """Add data into memory pool.
+
+        For rewards, we have a normal prior at first. After we observed a
+        reward for a given state-action pair, we use the mean value of our
+        observations instead of the prior mean as the posterior mean. The
+        standard deviations are in inverse proportion to the number of the
+        corresponding observations.
+
+        :param np.ndarray trans_count: the number of observations, with shape
+            (n_action, n_state, n_state).
+        :param np.ndarray rew_sum: total rewards, with shape
+            (n_state, n_action).
+        :param np.ndarray rew_count: the number of rewards, with shape
+            (n_state, n_action).
+        """
+        self.updated = False
+        self.trans_count += trans_count
+        sum_count = self.rew_count + rew_count
+        self.rew_mean = (self.rew_mean * self.rew_count + rew_sum) / sum_count
+        self.rew_std *= self.rew_count / sum_count
+        self.rew_count = sum_count
+        # I find that maybe we did not gather information
+        # for absorbing states' transition probabilities.
+        # After 1000 observations, if for some states, their
+        # trans_count almost do not change, the code adds
+        # the counts of these states transiting to themselves
+        # by 100.
+        if np.sum(self.trans_count) > \
+                np.sum(self.trans_count_prior_sum) + 1000:
+            min_index = np.argmin(np.sum(self.trans_count, axis=2), axis=1)
+            mask = np.isclose(np.sum(self.trans_count, axis=2),
+                              self.trans_count_prior_sum).astype("float32")
+            self.trans_count[np.array(range(self.n_action)),
+                             min_index, min_index] += \
+                mask[np.array(range(self.n_action)), min_index] * 100
+
+    def sample_from_prob(self) -> np.ndarray:
+        sample_prob = np.zeros_like(self.trans_count)
+        for i in range(self.n_action):
+            for j in range(self.n_state):
+                sample_prob[i][j] = np.random.dirichlet(
+                    self.trans_count[i][j])
+        return sample_prob
+
+    def sample_from_rew(self) -> np.ndarray:
+        sample_rew = np.random.randn(*self.rew_mean.shape)
+        sample_rew = sample_rew * self.rew_std + self.rew_mean
+        return sample_rew
```

Comment:
The current version is right. But maybe you can also use the np.random.normal to do the same thing?
First, it is consistent with the "sample_from_prob" function, second I think it will improve the readability.